### DISCUSSION TITLE
Can we install an older version of Oceananigans?

### DISCUSSION THREAD

**Discussion by weigwu** [CliMA/Oceananigans.jl#3078] [Computational science!] [answered]
Hi,

I've running a 3D eddy model GPU run for a while using Oceananigans. The experiment runs perfectly but recently I accidentally uninstall all my Julia packages. When I re-install the latest version Oceananigans and re-run my experiments, I got error messages when executing initial time step. A quick questions I want to ask before I re-debug everything is: is there a way to specify the version of Oceananigans to install in Julia? Any comments would be very vey helpful. 

Thanks,
Roger

(P.S. I also attached the error message I got
![Screenshot 2023-04-21 at 2 07 13 AM](https://user-images.githubusercontent.com/81963577/233553653-87ebac76-10d5-443c-8bad-8d671a911945.png)
)

**Comment by glwagner** [comment, marked as answer]
Yes, to install, for example, 0.79.0, you would type `]` to enter package manager mode, and then

```julia
pkg> add Oceananigans@0.79.0
```

You can use an environment using a `Project.toml` to avoid having to manually specify packages whenever you start using a new system.

I also suggest using github to store your code (which you may already do), along with the `Project.toml`. That will mean you could lose your whole computer but still regenerate your code, because the `Project.toml` can encode the "compatability" of your code with all the packages it uses.

Here's some documentation about that I recommend reading:

https://docs.julialang.org/en/v1/stdlib/Pkg/

https://pkgdocs.julialang.org/v1/environments/

**Reply by glwagner** [reply]
If you paste your whole code it might be easy to upgrade. I would need to see the entire error message to debug that... but reading the code might be easier depending on its complexity.

**Reply by weigwu** [reply]
Thanks a lot for these suggestions. I download version 0.76.0 that I used before. It works now. I will go ahead to setup Project.toml and store my codes. Thanks again for your help!
